src/index.ts

```diff
@@ -0,0 +1,282 @@
+    }
+    purchaserInfoUpdateListeners.push(purchaserInfoUpdateListener);
+  }
+
+  /**
+   * Removes a given PurchaserInfoUpdateListener
+   * @param {=PurchaserInfoUpdateListener} listenerToRemove =PurchaserInfoUpdateListener reference of the listener to remove
+   * @returns {Boolean} True if listener was removed, false otherwise
+   */
+  public static removePurchaserInfoUpdateListener(
+    listenerToRemove: PurchaserInfoUpdateListener
+  ) {
+    if (purchaserInfoUpdateListeners.includes(listenerToRemove)) {
+      purchaserInfoUpdateListeners = purchaserInfoUpdateListeners.filter(
+        listener => listenerToRemove !== listener
+      );
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Add a dict of attribution information
+   * @param {Dict} data Attribution data from AppsFlyer, Adjust, or Branch
+   * @param {ATTRIBUTION_NETWORKS} network Which network, see Purchases.ATTRIBUTION_NETWORKS
+   * @param {String?} networkUserId An optional unique id for identifying the user. Needs to be a string.
+   */
+  public static addAttributionData(
+    data: { [key: string]: any },
+    network: ATTRIBUTION_NETWORK,
+    networkUserId?: string
+  ) {
+    RNPurchases.addAttributionData(data, network, networkUserId);
+  }
+
+  /**
+   * Gets the map of entitlements -> offerings -> products
+   * @returns {Promise<Map<String, Map<String, Product>>>} Promise of entitlements structure
+   */
+  public static getEntitlements(): Promise<{
+    [entitlement: string]: { [productId: string]: any };
+  }> {
+    return RNPurchases.getEntitlements();
+  }
+
+  /**
+   * Fetch the product info
+   * @param {[String]} productIdentifiers Array of product identifiers
+   * @param {String} type Optional type of products to fetch, can be inapp or subs. Subs by default
+   * @returns {Promise<Array>} A promise containing an array of products. The promise will be rejected if the products are not properly
+   * configured in RevenueCat or if there is another error retrieving them. Rejections return an error code, and a userInfo object with more information.
+   */
+  public static getProducts(
+    productIdentifiers: string[],
+    type: string = "subs"
+  ) {
+    return RNPurchases.getProductInfo(productIdentifiers, type);
+  }
+
+  /**
+   * Make a purchase
+   * @param {String} productIdentifier The product identifier of the product you want to purchase
+   * @param {String?} oldSKU Optional sku you wish to upgrade from.
+   * @param {String} type Optional type of product, can be inapp or subs. Subs by default
+   * @returns {Promise<Object>} A promise of an object containing a purchaser info object and a product identifier. Rejections return an error code,
+   * a boolean indicating if the user cancelled the purchase, and an userInfo object with more information.
+   */
+  public static makePurchase(
+    productIdentifier: string,
+    oldSKU?: string | null,
+    type = "subs"
+  ) {
```

Comment:
Return type. It returns a Promise<PurchaserInfo> or throws the error. The error would contain another field "userCancelled", not sure if it's possible to type that...